app/src/main/java/pm/gnosis/heimdall/utils/AdapterUtils.kt

```diff
@@ -8,53 +8,82 @@ import pm.gnosis.heimdall.common.utils.ErrorResult
 import pm.gnosis.heimdall.common.utils.Result
 import pm.gnosis.heimdall.ui.base.Adapter
 
+fun <D> Flowable<List<D>>.scanToAdapterData(idExtractor: ((D) -> Any) = defaultIdExtractor(), payloadCalc: ((D, D) -> Any?)? = null): Flowable<Adapter.Data<D>> =
```

Comment:
I am not sure if the idExtractor or itemCheck was better. Any preferences?

Reply:
Code with idExtractor tends to be shorter. But you can do more with the itemCheck (as you could compare more than just an id)

Reply:
Fine with both 👍